### DISCUSSION TITLE
`Hono.prototype.route` loses schema information gained by `TypedResponse`

### DISCUSSION THREAD

**Discussion by yudai-nkt** [honojs/hono#907] [Q&A]
I'm trying `TypedResponse` but can't figure out how to use it properly with the `route` method. Consider the following example:

```ts
import { Hono } from "hono";

const blog = new Hono().get("/posts", (c) =>
  c.jsonT([{ slug: "congrats-on-v3-release" }])
);
const app = new Hono().route("/blog", blog);

export default app;
```

In the above code, `blog` has a type of `Hono<Env, Schema<"get", "/posts", {}, { slug: string }[]>>` but `app`'s type is `Hono<Env, {}>`. I want it to be `Hono<Env, Schema<"get", "/blog/posts", {}, { slug: string }[]>>`.

How can I keep the schema information, or is this something that isn't supported yet? If the latter is the case, I'll convert this discussion into an issue.

**Comment by yusukebe** [comment]
HI @yudai-nkt !

Thank you for trying.

I thought it was not too bad if the schema information disappeared, but it should keep it.

> I'll convert this discussion into an issue.

Please. I may fix it.

**Comment by yusukebe** [comment]
Hey, I've created the PR #909. It may fix this issue.

**Reply by yudai-nkt** [reply]
wow, thanks for the fix and sorry for being late. i'll check and try the branch when i get back home!
